util/cmake/SeqAnBuildSystem.cmake

```diff
@@ -197,68 +197,68 @@ macro (seqan_build_system_init)
          ${PROJECT_BINARY_DIR}/bin)
 
     # Set Warnings
-    if (CMAKE_COMPILER_IS_GNUCXX OR COMPILER_IS_CLANG OR COMPILER_IS_INTEL)
-        set (CMAKE_CXX_WARNING_LEVEL 4)
+    if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+        # Disable warnings about unsecure (although standard) functions.
+        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} /W2 /D_SCL_SECURE_NO_WARNINGS") # TODO(h-2): raise this to W4
+    else ()
         set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -W -Wall -pedantic -fstrict-aliasing -Wstrict-aliasing")
         set (SEQAN_DEFINITIONS ${SEQAN_DEFINITIONS} -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64)
-    endif ()
-    if (COMPILER_IS_INTEL)
-        # disable some warnings on ICC
-        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -wd3373,2102")
-    endif (COMPILER_IS_INTEL)
-    if (MSVC)
-        # Use the /W2 warning level for visual studio.
-        set (CMAKE_CXX_WARNING_LEVEL 2) # TODO(h-2): raise this to W4
-        # Disable warnings about unsecure (although standard) functions.
-        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} /D_SCL_SECURE_NO_WARNINGS")
+        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+            # disable some warnings on ICC
+            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -wd3373,2102")
+        endif ()
     endif ()
 
-    # DEFFAULT BUILD SYSTEM
     if (NOT SEQAN_BUILD_SYSTEM)
         set (SEQAN_BUILD_SYSTEM "DEVELOP" CACHE STRING "Build/Release mode to select. One of DEVELOP SEQAN_RELEASE, APP:\${APP_NAME}. Defaults to DEVELOP.")
     endif (NOT SEQAN_BUILD_SYSTEM)
     set (SEQAN_APP_VERSION "0.0.0" CACHE STRING "Version of the application.")
     set (SEQAN_NIGHTLY_RELEASE FALSE CACHE BOOL "Set to TRUE to enable nightly app releases.")
 
-    # STANDARD BUILD FLAGS
-
-    # OpenBSD just can't handle it
-    if (${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
-        set (SEQAN_NO_NATIVE TRUE)
-        set (SEQAN_OFFICIAL_PKGS FALSE)
+    ## options
+    if (NOT SEQAN_64BIT_TARGET_PLATFORM)
+        set (SEQAN_ARCH_SSE4 FALSE)
     endif ()
 
-    # packages for distribution
-    if ((SEQAN_OFFICIAL_PKGS) AND
-        (NOT MSVC) AND
-        (SEQAN_BUILD_SYSTEM MATCHES "APP")) # either APP:$app or SEQAN_RELEASE_APPS
-
-        # static linkage
-        set (SEQAN_STATIC_APPS TRUE CACHE INTERNAL "Create static app binaries")
-        message (STATUS "Building static binaries.")
+    if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+        set (SEQAN_STATIC_APPS FALSE)
+        set (SEQAN_ARCH_SSE4 FALSE)
```

Comment:
I am not sure these things exist in that way for windows. But if there as VS flags that work, please add them :)
[but you could do it in a seperate PR as it is new functionality anyways]